### PR TITLE
Add motion.set_support_self + tests.

### PR DESCRIPTION
**Pull request by reiterl**
Added checks.
Added support add/remove code.
Only use instance if something changed.
Added tests.

### CODE REVIEW

File: tests/system/action/motion/test_set_support_self.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            json=[
+                {
+                    "action": "motion.set_support_self",
+                    "data": [{"motion_id": 1, "support": True}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        model = self.get_model("motion/1")
+        assert model.get("supporter_ids") == [1]
```
Comment (jsangmeister) [inline tests/system/action/motion/test_set_support_self.py:109]: Please assert that the reverse relation is set correctly.

File: tests/system/action/motion/test_set_support_self.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            json=[
+                {
+                    "action": "motion.set_support_self",
+                    "data": [{"motion_id": 1, "support": False}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        model = self.get_model("motion/1")
+        assert model.get("supporter_ids") == []
```
Comment (jsangmeister) [inline tests/system/action/motion/test_set_support_self.py:153]: Please assert that the reverse relation is set correctly.

File: openslides_backend/action/actions/motion/set_support_self.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                ["motions_supporters_min_amount"],
+            )
+            if meeting.get("motions_supporters_min_amount") == 0:
+                raise ActionException("Motion supporters system deactivated.")
+            if not motion.get("state_id"):
+                raise ActionException("Motion is missing state_id.")
+            state = self.datastore.get(
+                FullQualifiedId(Collection("motion_state"), motion["state_id"]),
+                ["allow_support"],
+            )
```
Comment (jsangmeister) [inline openslides_backend/action/actions/motion/set_support_self.py:46]: Please bundle these in as few datastore calls as possible: One to fetch all motions and one to fetch all meetings and states should be enough (so 2 calls total).

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: tests/system/action/motion/test_set_support_self.py (hunk 1 above)
Comment (reiterl) [reply tests/system/action/motion/test_set_support_self.py:109]: Added a reverse relation check.

File: tests/system/action/motion/test_set_support_self.py (hunk 2 above)
Comment (reiterl) [reply tests/system/action/motion/test_set_support_self.py:153]: Added a reverse relation check.

File: openslides_backend/action/actions/motion/set_support_self.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/actions/motion/set_support_self.py]: Use Get Many Requests to reduce to only two datastore calls.

File: tests/system/action/motion/test_set_support_self.py (hunk 1 above)
Comment (jsangmeister) [reply tests/system/action/motion/test_set_support_self.py:119]: Also check that the template field is set correctly (`supported_motion_$_ids == ["1"]`)

File: tests/system/action/motion/test_set_support_self.py (hunk 2 above)
Comment (jsangmeister) [reply tests/system/action/motion/test_set_support_self.py:165]: same as above

**Review by jsangmeister — CHANGES_REQUESTED** [review]

**Comment by reiterl** [comment]
Add checks for the template field.

**Review by jsangmeister — COMMENTED** [review]
looks good. Some tests are failing which is loosely related to https://github.com/OpenSlides/openslides-backend/issues/365 (Background: `equal_fields` does not work for users since the y do not necessarily have a `meeting_id`). I will take care of this.

**Pull request merged**